bin/deploy

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+
+main() {
+  sha=`git rev-parse --short HEAD`
+
+  create_new_branch
+  commit_gemfile_lock
+  push_branch_to_staging
+}
+
+create_new_branch() {
+  local user_name=`git config user.name |
+  cut -d ' ' -f 1 |
+  tr '[:upper:]' '[:lower:]'`
+  branch_name="staging-$sha-$user_name"
+
+  git checkout -b $branch_name
+}
+
+commit_gemfile_lock() {
+  git add -f Gemfile.lock
+  git commit -m "Deploying $sha"
+}
+
+push_branch_to_staging() {
+  git push --force staging $branch_name:master
+}
+
+main
```

Comment:
For sh, I'd rather see things listed out instead of abstracted. Like:

```sh
#!/bin/sh

sha=`git rev-parse --short HEAD`
user_name=`git config user.name | \
  cut -d ' ' -f 1 | \
  tr '[:upper:]' '[:lower:]'`
branch_name="staging-$sha-$user_name"

git checkout -b $branch_name
git add -f Gemfile.lock
git commit -m "Deploying $sha"
git push --force-with-lease staging $branch_name:master
```